### ISSUE TITLE
False positives due to lack of constant propagation

### ISSUE THREAD

**Issue by topolarity** [tz-lom/TrimCheck.jl#15]
It is possible (in some cases, quite likely) to see false positives due to checking the type signature of a call, without constant arguments:
```julia
julia> TrimCheck.validate(:(hvncat(Int, Vector{Float64}, Vector{Float64}, Vector{Float64})))
Validating call: hvncat(Int, Vector{Float64}, Vector{Float64}, Vector{Float64})
Done ✓
Validation result #1
Call `hvncat(Int, Vector{Float64}, Vector{Float64}, Vector{Float64})` failed trim checking
```

The intent of the user may be to test a call like:
```julia
hvncat(2, randn(3), randn(3), randn(3))
```

since this corresponds to the syntax `[randn(3) ;; randn(3) ;; randn(3)]` and the constant `2` is inserted by the compiler

Since `Base.hvncat` is [labelled with `@constprop :aggressive`](https://github.com/JuliaLang/julia/blob/6bd20ed92ac1e4cb41d50d4ee05775819eeb9683/base/abstractarray.jl#L2387-L2388), it makes a significant difference to type stability when the `2` is known to be a `Const(2)` or if it is instead treated merely as `Int`.

One possible workaround is to wrap the function in a zero-argument closure, so that the constant is visible to the compiler:
```julia
julia> TrimCheck.validate(:((()->hvncat(2, rand(3), rand(3), rand(3)))()))
Validating call: ((()->begin
        #= REPL[8]:1 =#
        hvncat(2, rand(3), rand(3), rand(3))
    end))()
Done ✓
Validation result #1
Call `((()->begin
        #= REPL[8]:1 =#
        hvncat(2, rand(3), rand(3), rand(3))
    end))()` is trim compatible.
```
